src/connectors/infinite-hits/connectInfiniteHits.ts

```diff
@@ -242,20 +247,11 @@ const connectInfiniteHits: InfiniteHitsConnector = (
           );
         }
 
-        if (!hasShowPrevious) {
-          return widgetSearchParameters;
-        }
-
-        if (uiState.page) {
-          // The page in the search parameters is decremented by one
-          // to get to the actual parameter value from the UI state.
-          return widgetSearchParameters.setQueryParameter(
-            'page',
-            uiState.page - 1
-          );
-        }
+        // The page in the search parameters is decremented by one
+        // to get to the actual parameter value from the UI state.
+        const page = hasShowPrevious && uiState.page ? uiState.page - 1 : 0;
```

Comment:
Updated the condition to get the current page, should be more coherent now.

For the behavior when we remove a widget, the main issue is that we do not reset the page when a widget with some refinements is removed. I don't think this is related to the infinite hits in particular (even if the behavior is more problematic for it since we cannot go back if `showPrevious` is disabled).

Reply:
Updated the condition to get the current page, should be more coherent now.

For the behavior when we remove a widget, the main issue is that we do not reset the page when a widget with some refinements is removed. I don't think this is related to the infinite hits in particular (even if the behavior is more problematic for it since we cannot go back if showPrevious is disabled).

Reply:
Yes it's not related but as you said it's a bit more problematic with `infiniteHits`. At the moment the behavior is a bit weird but nobody complains about that in React InstantSearch (we have the same behavior). We can think of a proper solution later.